pytorch_lightning/trainer/_progress.py

```diff
@@ -0,0 +1,113 @@
+    These counters are local to a trainer rank.
+    By default, they are not by default globally synced across all ranks.
+
+    Args:
+        total_epochs_finished: Number of passes through the dataset.
+        total_batches_read: Number of batches loaded through the dataloader.
+        total_batches_started: Number of batches started processing.
+            This is intended to be incremented after `on_train_batch_start` completes.
+        total_batches_processed: Number of batches processed.
+            This is intended to be incremented after `training_step` runs.
+        total_batches_finished: Number of batches finished.
+            This is intended to be incremented after `on_train_batch_end` runs.
+
+        batches_read_this_epoch: Number of batches loaded through the dataloader within the current epoch.
+        batches_started_this_epoch: Number of batches started processing within the current epoch.
+            This is intended to be incremented after `on_train_batch_start` completes.
+        batches_processed_this_epoch: Number of batches processed within the current epoch.
+            This is intended to be incremented after `training_step` runs.
+        batches_finished_this_epoch: Number of batches finished within the current epoch.
+            This is intended to be incremented after `on_train_batch_end` runs.
+    """
+    total_epochs_finished: int = 0
+
+    total_batches_read: int = 0
+    total_batches_started: int = 0
+    total_batches_processed: int = 0
+    total_batches_finished: int = 0
+
+    batches_read_this_epoch: int = 0
+    batches_started_this_epoch: int = 0
+    batches_processed_this_epoch: int = 0
+    batches_finished_this_epoch: int = 0
+
+    def increment_batch_read(self) -> None:
+        self.total_batches_read += 1
+        self.batches_read_this_epoch += 1
+
+    def increment_batch_started(self) -> None:
+        self.total_batches_started += 1
+        self.batches_started_this_epoch += 1
+
+    def increment_batch_processed(self) -> None:
+        self.total_batches_processed += 1
+        self.batches_processed_this_epoch += 1
+
+    def increment_batch_finished(self) -> None:
+        self.total_batches_finished += 1
+        self.batches_finished_this_epoch += 1
+
+    def increment_epoch_finished(self) -> None:
+        self.total_epochs_finished += 1
+        self.reset_on_epoch()
+
+    def reset_on_epoch(self) -> None:
+        self.batches_read_this_epoch = 0
+        self.batches_started_this_epoch = 0
+        self.batches_processed_this_epoch = 0
+        self.batches_finished_this_epoch = 0
+
+
+@dataclass
+class TrainLoopProgress(LoopProgress):
+    """Extension of ``LoopProgress`` for training specific fields.
+
+    Optimizer steps (parameter updates) are unique to training.
+    """
+
+    total_optimizer_steps_processed: int = 0
+    optimizer_steps_processed_this_epoch: int = 0
```

Comment:
```suggestion
    total_optimizer_steps_processed: int = 0
    optimizer_steps_processed_this_epoch: int = 0
```
Do we want more consistent naming structure and go for

`optimizer_steps_processed_total`?

easier to memorize if there is a pattern to recognize imo.
on the other hand, you have total prefixed everywhere else ^^

Reply:
I agree.

Also `this_epoch` -> `epoch`?

Reply:
@awaelchli @carmocca so: `optimizer_steps_processed_total` and `optimizer_steps_processed_epoch` ?

Reply:
yes